tsdb/index/inmem/inmem.go

```diff
@@ -189,13 +189,23 @@ func (i *Index) MeasurementIterator() (tsdb.MeasurementIterator, error) {
 func (i *Index) CreateSeriesListIfNotExists(seriesIDSet *tsdb.SeriesIDSet, measurements map[string]int,
 	keys, names [][]byte, tagsSlice []models.Tags, opt *tsdb.EngineOptions, ignoreLimits bool) error {
 
+	// Verify that the series will not exceed limit.
+	if !ignoreLimits {
+		i.mu.RLock()
```

Comment:
seems like there might be a race between here and line 201 where the series might exceed the limit after this check.

it seems like everything between lines 192 and 212 should be in a read lock?

Reply:
See response above.